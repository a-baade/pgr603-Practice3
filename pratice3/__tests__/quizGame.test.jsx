import React from "react";
import ReactDom from "react-dom";
import {FrontPage, QuizGame, QuestionContext, ShowQuestion} from "../quizGame";
import {MemoryRouter} from "react-router-dom";
import {Simulate} from "react-dom/test-utils";
import pretty from "pretty";

const question = {
    question: "How are you?",
    answers: {
        answer_a: "Good",
        answer_b: "Bad",
        answer_c: "OK",
    },
    correct_answers: {
        answer_a_correct: "true",
        answer_b_correct: "false",
        answer_c_correct: "false",
    }
}

describe("Quiz game", () => {

    it("Shows answer status",() => {
        const element = document.createElement("div");
        ReactDom.render(
            <MemoryRouter><FrontPage correctAnswers={3} questionsAnswered={10} /></MemoryRouter>,
            element
        );
        expect(element.querySelector("[data-testid=status]").textContent)
            .toEqual("You have answered 3 of 10 correctly");
        expect(pretty(element.innerHTML)).toMatchSnapshot();
    })
    it("Shows question", () => {
        const question = {
            question: "How are you?",
            answers: {
                answer_a: "Good",
                answer_b: "Bad",
                answer_c: "OK",
            }
        }
        const element = document.createElement("div");
        ReactDom.render(
            <MemoryRouter initialEntries={["/question"]}>
                <QuestionContext.Provider value={{randomQuestion: () => question}}>
                    <QuizGame/>
                </QuestionContext.Provider>
                </MemoryRouter>,
        element
        );
        expect(pretty(element.innerHTML)).toMatchSnapshot();
    })
    it("records correct answer", () => {

        const setQuestionsAnswered = jest.fn()
        const setCorrectAnswers = jest.fn()

        const element = document.createElement("div");
        ReactDom.render(
            <MemoryRouter initialEntries={["/question"]}>
                <QuestionContext.Provider value={{randomQuestion: () => question}}>
                    <ShowQuestion setCorrectAnswers={setCorrectAnswers} setQuestionsAnswered={setQuestionsAnswered}/>
                </QuestionContext.Provider>
            </MemoryRouter>,
            element
        );
        Simulate.click(element.querySelector("[data-testid=answer_a] button"));
        expect(setQuestionsAnswered).toBeCalled();
        expect(setCorrectAnswers).toBeCalled();
        expect(pretty(element.innerHTML)).toMatchSnapshot();
    });
})