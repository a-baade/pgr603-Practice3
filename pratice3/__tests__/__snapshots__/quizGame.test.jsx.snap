// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz game Shows answer status 1`] = `
"<div>
  <h1>Quiz App</h1>
  <div data-testid=\\"status\\">You have answered 3 of 10 correctly</div><a href=\\"/question\\"><button>Take a quiz</button></a>
</div>"
`;

exports[`Quiz game Shows question 1`] = `
"<div>
  <h1>How are you?</h1>
  <div data-testid=\\"answer_a\\"><button>Good</button></div>
  <div data-testid=\\"answer_b\\"><button>Bad</button></div>
  <div data-testid=\\"answer_c\\"><button>OK</button></div>
</div>"
`;

exports[`Quiz game records correct answer 1`] = `
"<div>
  <h1>How are you?</h1>
  <div data-testid=\\"answer_a\\"><button>Good</button></div>
  <div data-testid=\\"answer_b\\"><button>Bad</button></div>
  <div data-testid=\\"answer_c\\"><button>OK</button></div>
</div>"
`;
